src/ORM/DataQuery.php

```diff
@@ -442,14 +443,28 @@ protected function ensureSelectContainsOrderbyColumns($query, $originalSelect =
         }
     }
 
+    /**
+     * Calls a callback on either the primary database or a replica, with respect to the configured
+     * value of `must_use_primary_db` on the current dataClass
+     */
+    private function withCorrectDatabase(callable $callback): mixed
```

Comment:
Move the private method to the bottom of the class per https://docs.silverstripe.org/en/5/contributing/coding_conventions/php_coding_conventions/#class-member-ordering

Reply:
Updated